cmd/frontend/internal/modelconfig/siteconfig.go

```diff
@@ -0,0 +1,312 @@
+package modelconfig
+
+import (
+	"fmt"
+
+	"github.com/sourcegraph/log"
+
+	"github.com/sourcegraph/sourcegraph/internal/conf"
+	"github.com/sourcegraph/sourcegraph/internal/modelconfig/types"
+	"github.com/sourcegraph/sourcegraph/schema"
+)
+
+// getSiteModelConfigurationOrNil returns the SiteModelConfiguration, or nil if there is none.
+//
+// This function is responsible for converting the schema.SiteConfiguration that admins write
+// into our internal data type.
+func getSiteModelConfigurationOrNil(logger log.Logger, siteConfig schema.SiteConfiguration) (*types.SiteModelConfiguration, error) {
+	// If "modelConfiguration" is specified, then we respect that and only that. If it is not specified,
+	// then we respect the older "completions" configuration.
+	modelConfig := siteConfig.ModelConfiguration
+	if modelConfig == nil {
+		if compConfig := conf.GetCompletionsConfig(siteConfig); compConfig != nil {
+			logger.Info("converting completions configuration data", log.String("apiProvider", string(compConfig.Provider)))
+			return convertCompletionsConfig(compConfig)
+		}
+		return nil, nil
+	}
+	return convertModelConfiguration(modelConfig), nil
```

Comment:
It seems like we could clean this up considerably by just having `if modelConfig != nil { return convertModelConfiguration` } if `compConfig != nil { return convertCompletionsConfig`.

Right?

```suggestion

	if modelConfig := siteConfig.ModelConfiguration; modelConfig != nil {
	    logger.Info("converting model configuration data")
	    return convertModelConfiguration(modelConfig), nil
	}
	
	if compConfig := conf.GetCompletionsConfig(siteConfig); compConfig != nil {
			logger.Info("converting completions configuration data", log.String("apiProvider", string(compConfig.Provider)))
			return convertCompletionsConfig(compConfig)
		}

	return nil, errors.New("no Cody configuration data available")
```

Reply:
I can invert the conditional as you mentioned, yeah.

There is a problem with your suggested code snippet, though:

```
return nil, errors.New("no Cody configuration data available")
```

This function does need to return `nil, nil` in this case because there is no guarantee a Sourcegraph instance has either "completions" or "modelConfiguration" in their site config ('not using Cody' case)

Reply:
Addressed in a8b9df60345